src/lib.rs

```diff
@@ -22,6 +22,23 @@
 )]
 #![doc(html_root_url = "https://docs.rs/curve25519-dalek/4.0.0-pre.2")]
 #![doc = include_str!("../README.md")]
+//------------------------------------------------------------------------
+// curve25519-dalek arithmetric default and overrides
+//------------------------------------------------------------------------
+#![cfg_attr(
+    any(
+        all(not(target_pointer_width = "64"), not(dalek_bits = "64")),
+        dalek_bits = "32"
+    ),
+    cfg(dalek_bits = "32")
+)]
+#![cfg_attr(
+    any(
+        all(target_pointer_width = "64", not(dalek_bits = "32")),
+        dalek_bits = "64"
+    ),
+    cfg(dalek_bits = "64")
+)]
```

Comment:
Perhaps this could be addressed in a followup PR, but it seems there are a few `target_arch`es wwhere `target_pointer_width = "32"` where we might want to override the default 32-backend, namely `wasm`

Reply:
This default / override logic is now in build.rs as this does not work via cfg_attr - naive / tired I again was :woman_shrugging: 

Nonetheless yeah definitely it would be nice to map those overrides out

Reply:
Raised an issue: https://github.com/dalek-cryptography/curve25519-dalek/issues/456